### PR TITLE
Fix old sbcc and make old/new sbcc co-exist

### PR DESCRIPTION
**Pull request by solaslin**
back port from latest branch: switch len 100 to old SBCC

### CODE REVIEW

**Review by evetsso — APPROVED** [review]

**Review by memmett — APPROVED** [review]

**Review by feizheng10 — APPROVED** [review]

File: library/src/device/generator/generator.kernel.hpp [hunk 1]
```diff
@@ -1158,7 +1158,7 @@ namespace StockhamGenerator
             // Function signature
             if(NeedsLargeTwiddles())
             {
-                str += "template <typename T, StrideBin sb, bool TwdLarge, CallbackType cbtype, "
+                str += "template <typename T, StrideBin sb, CallbackType cbtype, bool TwdLarge, "
```
Comment (malcolmroberts) [inline library/src/device/generator/generator.kernel.hpp:1161]: This was switched for some reason?

File: library/src/plan.cpp [hunk 2]
```diff
@@ -1007,21 +1007,41 @@ bool TreeNode::use_CS_2D_SINGLE()
...
-    //   However, technically no LDS limitation along the fast dimension
-    //   on upper bound for 2D SBCC cases, and even should not limit to pow
-    //   of 2.
-
-    // FIXME: use all SBCC kernels instead after we fix the bugs in buffer assignment
-    // std::set<int> sbcc_support = {50, 64, 81, 100, 128, 200, 256};
-    std::set<int> sbcc_support = {64, 128, 256};
-    if((sbcc_support.find(length[1]) != sbcc_support.end()) && (length[0] >= 64))
+    try
+    {
```
Comment (malcolmroberts) [inline library/src/plan.cpp:1011]: This looks like we should expect the try block to fail under normal circumstances.  Using exceptions for normal flow seems like a bad idea; exceptions should be for exceptional circumstances.

File: library/src/device/generator/generator.kernel.hpp (hunk 1 above)
Comment (evetsso) [reply library/src/device/generator/generator.kernel.hpp:1161]: I put cbtype before TwdLarge when callbacks were merged.  But that was after sbcc was moved to the new generator, so the old generator wasn't updated.  This change is updating the old generator to match what the new generator is now doing.

File: library/src/plan.cpp (hunk 2 above)
Comment (evetsso) [reply library/src/plan.cpp:1011]: Rearranging that should be the subject of a different PR, though.

**Comment by malcolmroberts** [comment]
Is there any impact on other tests?  Please run the performance suite on MI50 and MI100.

**Comment by memmett** [comment]
It's 5am in Taiwan.  I can run on MI100...

**Comment by malcolmroberts** [comment]
MI50 looks good.
[figs.pdf](https://github.com/ROCmSoftwarePlatform/rocFFT/files/6678490/figs.pdf)

**Pull request merged**